cirq-core/cirq/circuits/moment_test.py

```diff
@@ -294,6 +294,11 @@ def test_resolve_parameters():
     moment = cirq.Moment(cirq.X(a) ** sympy.Symbol('v'), cirq.Y(b) ** sympy.Symbol('w'))
     resolved_moment = cirq.resolve_parameters(moment, cirq.ParamResolver({'v': 0.1, 'w': 0.2}))
     assert resolved_moment == cirq.Moment(cirq.X(a) ** 0.1, cirq.Y(b) ** 0.2)
+    # New test for empty resolver dictionary
+    empty_moment = cirq.Moment(cirq.X(a))
+    resolved_empty_moment = cirq.resolve_parameters(empty_moment, {})
+    assert resolved_empty_moment == empty_moment
+
```

Comment:
This does not seem to verify the fix, because it passes in the main branch.
Also using an empty dictionary in `cirq.resolve_parameters` is a noop, ie, equivalent to
`resolved_empty_moment = empty_moment`.

Please replace with

```suggestion
    # sympy constant is resolved to a Python number
    moment = cirq.Moment(cirq.Rz(rads=sympy.pi).on(a))
    resolved_moment = cirq.resolve_parameters(moment, {'pi': np.pi})
    assert resolved_moment == cirq.Moment(cirq.Rz(rads=np.pi).on(a))
    resolved_gate = resolved_moment.operations[0].gate
    assert not isinstance(resolved_gate.exponent, sympy.Basic)
    assert isinstance(resolved_gate.exponent, float)
    assert not cirq.is_parameterized(resolved_moment)
```